cloudinit/sources/DataSourceAzure.py

```diff
@@ -792,6 +793,7 @@ def activate(self, cfg, is_new_instance):
         address_ephemeral_resize(is_new_instance=is_new_instance,
                                  preserve_ntfs=self.ds_cfg.get(
                                      DS_CFG_KEY_PRESERVE_NTFS, False))
+        push_log_to_kvp()
```

Comment:
If address_ephemeral_resize has an exception, the call to push_log_to_kvp will get skipped.
Perhaps something like this:
`try:
       address_ephemeral_resize(is_new_instance=is_new_instance,
                                 preserve_ntfs=self.ds_cfg.get(
                                     DS_CFG_KEY_PRESERVE_NTFS, False))
finally:
        push_log_to_kvp()`